### DISCUSSION TITLE
Documentation for the rich text editor

### DISCUSSION THREAD

**Discussion by mazzelman** [pocketbase/pocketbase#4745] [Q&A] [answered]
Hello,

i searched the internet, and the closed issues, but couldn't find an answer. I was hoping, that my question can be answered from anyone more experienced then me. I'm trying to build my own little blog app with Pocketbase and NextJS. I really like the Pocketbase backend. I setup a collection of posts and i can get the list of all posts in NextJS. My problem is, that when i map over the posts the html tags are within the code. The terminal output looks like this:

`Images: [],
  Text: '<p>The awesome cat place... is just awesome!</p>',
  Title: 'The cat place',
  Videos: [],` 

How can i map over the posts and display my created textfield? I read something about "raw" but i think this would "trim" the html tags out. It would be really cool to use the formatting from the **rich field text editor**. In my posts page i use this code to get the posts:

```
import { getPosts } from "@/lib/connect";

export default async function Posts() {
  const posts = await getPosts();

  return (
    <>
      <h1>Posts</h1>
      {posts.map((post) => (
        <div key={post.id}>
          <h2>{post.Title}</h2>
          {post.Text}
        </div>
      ))}
    </>
  );
}
```

Thanks Marcel 🖖

**Comment by ganigeorgiev** [comment, marked as answer]
I don't understand what actually is the question and what documentation you are looking for.

The `editor` type field returns its content as plain HTML string. If you want to render the HTML as part of your react component you can check the [`dangerouslySetInnerHTML` React prop](https://react.dev/reference/react-dom/components/common#dangerously-setting-the-inner-html), but as the name of the prop suggest use it only with trusted input or "purify" it with [`dompurify`](https://github.com/cure53/DOMPurify) or other similar library (_there could be also a React wrapper for it_).

**Reply by mazzelman** [reply]
Hi,
thanks for the answer. I thought there would be a documentation how i should use it. But your links point me in the right direction! I choose the "dangerouslySetInnerHTML". The code looks now like this now:

```
import { getPosts } from "@/lib/connect";

export default async function Posts() {
  const posts = await getPosts();

  return (
    <>
      <h1>Posts</h1>
      {posts.map((post) => (
        <div key={post.id}>
          <h2>{post.Title}</h2>
          <div dangerouslySetInnerHTML={{ __html: post.Text }} />
        </div>
      ))}
    </>
  );
}
```
The only one who put text in this pocketbase instance is me. So i think this would be okay. I will take a look at dompurify too. This article helped me too:

[https://sentry.io/answers/using-dangerouslysetinnerhtml-in-next-js/](https://sentry.io/answers/using-dangerouslysetinnerhtml-in-next-js/)

I would like to apologize if my post caused any confusion. I will make sure to post my questions in the right place in the future.

Thank you, for your time and help! 👍
